pkg/archive/archive.go

```diff
@@ -460,6 +460,20 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 			return err
 		}
 	}
+
+	if hdr.Typeflag == tar.TypeReg || hdr.Typeflag == tar.TypeRegA {
+		file, err := os.Open(path)
+		if err != nil {
+			file.Close()
```

Comment:
@willnewton, Actually I prefer it as it's done in the original PR. Doing `defer` in a file traversal procedure which has recursive nature could be very tricky. As you you know there is a LIFO queue that accumulates all the defer jobs, those are executed only after you leave the function,  meanwhile you can easily exceed the ulimit of concurrently opened file descriptors.